### PR TITLE
Refactor `teams.php`

### PR DESCRIPTION
**Pull request by everly-gif**
## What does this PR do?

Refactors and cleans up code in `teams.php`

## Test Plan

Manual 
- Checked if the IDE is still able to pickup/identify the params.

## Related PRs and Issues

NIL

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: app/controllers/api/teams.php [hunk 1]
```diff
@@ -42,11 +45,7 @@
     ->inject('user')
     ->inject('dbForProject')
     ->inject('events')
-    ->action(function ($teamId, $name, $roles, $response, $user, $dbForProject, $events) {
-        /** @var Appwrite\Utopia\Response $response */
-        /** @var Utopia\Database\Document $user */
-        /** @var Utopia\Database\Database $dbForProject */
-        /** @var Appwrite\Event\Event $events */
+    ->action(function ($teamId, $name, $roles, Response $response, Document $user, Database $dbForProject, Event $events) {
```
Comment (eldadfux) [inline app/controllers/api/teams.php:48]:
```suggestion
    ->action(function (string $teamId, string $name, array $roles, Response $response, Document $user, Database $dbForProject, Event $events) {
```

I think we can also add primitive types to params we get from the http server.

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: app/controllers/api/teams.php (hunk 1 above)
Comment (everly-gif) [reply app/controllers/api/teams.php:48]: Sure ! I'll make the changes

File: app/controllers/api/teams.php [hunk 2]
```diff
@@ -281,13 +270,7 @@
...
     ->inject('audits')
     ->inject('mails')
-    ->action(function ($teamId, $email, $roles, $url, $name, $response, $project, $user, $dbForProject, $locale, $audits, $mails) {
-        /** @var Appwrite\Utopia\Response $response */
-        /** @var Utopia\Database\Document $project */
-        /** @var Utopia\Database\Document $user */
-        /** @var Utopia\Database\Database $dbForProject */
-        /** @var Appwrite\Event\Event $audits */
-        /** @var Appwrite\Event\Event $mails */
+    ->action(function (string $teamId, string $email, array $roles, string $url, string $name, Response $response, Document $project, Document $user, Database $dbForProject, $locale, Event $audits, Event $mails) {
```
Comment (lohanidamodar) [inline app/controllers/api/teams.php:273]:
```suggestion
    ->action(function (string $teamId, string $email, array $roles, string $url, string $name, Response $response, Document $project, Document $user, Database $dbForProject, Locale $locale, Event $audits, Event $mails) {
```
`$locale` is `Utopia\Locale\Locale`

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Small change, rest looks good.

File: app/controllers/api/teams.php (hunk 2 above)
Comment (everly-gif) [reply app/controllers/api/teams.php]: Sure, I'll make the changes.

**Review by lohanidamodar — APPROVED** [review]
LGTM!

**Review by eldadfux — APPROVED** [review]

**Comment by eldadfux** [comment]
Great stuff, code looks way cleaner 👍

**Pull request merged**